### PR TITLE
Loki: Fix text search in Label browser

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
In the future, we plan to [implement proper fuzzy search](https://github.com/grafana/grafana/issues/32209). In the meantime for 7.5, we have decided to improve the current experience and implement simple text search.

### CODE REVIEW

**Comment by aocenas** [comment]
could there be any issue with special characters? Like could there `.` or `\` in the loki labels? That would mean you have to escape them in the search right? Also can we change the label to say it expects a regex? Otherwise it's hard to discover (probably only by seeing it did not match a special char a then figuring out why)

**Comment by davkal** [comment]
> could there be any issue with special characters?

I think this could mess with IPs, where `1.2.3` would also match `192.3`. Perhaps we make the regexp explicit by adding a checkbox with "Use as regex" next to the input field. But in the interest of time, perhaps we should just use text matching for now.

**Comment by ivanahuckova** [comment]
Yeah, good catch and good point @aocenas. @davkal I have switched this to simple text search.

File: public/app/plugins/datasource/loki/components/LokiLabel.tsx [hunk 1]
```diff
@@ -24,6 +24,7 @@ export const LokiLabel = forwardRef<HTMLElement, Props>(
   ({ name, value, hidden, facets, onClick, className, loading, searchTerm, active, style, ...rest }, ref) => {
     const theme = useTheme();
     const styles = getLabelStyles(theme);
+    const matcher = searchTerm ? new RegExp(searchTerm, 'i') : undefined;
```
Comment (ifrost) [inline public/app/plugins/datasource/loki/components/LokiLabel.tsx:27]: Nit: Highlighter supports passing strings so there's no need to convert it a RegExp.

**Review by ifrost — APPROVED** [review]

**Pull request merged**